trade-finance-manager-ui/server/controllers/utilisation-reports/index.test.ts

```diff
@@ -88,7 +88,6 @@ describe('controllers/utilisation-reports', () => {
       await getUtilisationReports(req, res);
 
       // Assert
-      expect(isTfmPaymentReconciliationFeatureFlagEnabledSpy).toHaveBeenCalledTimes(1);
```

Comment:
It actually gets called twice now but that feels like more of an implementation detail rather than something we explicitly want to assert happens

Reply:
I thought that would be why - just wanted to confirm!

Reply:
Sorry, just stumbled across this - I don't understand why we would _not_ want to test this?

I think it should remain as is because with removing it - the flag could then be removed/disabled in the controller and the test would still pass.

If it's getting called twice, it sounds like the mock/spy needs to be reset or setup inbetween tests

Reply:
The clean up is working correctly, the logic has been changed in the controller so we do genuinely call the helper twice (through two separate helper methods used by the page).

> the flag could then be removed/disabled in the controller and the test would still pass.

That is actually why we don't need this assertion, if the check gets removed and the assertion below still passes, then the behaviour is still the desired behaviour, since the assertion below this one is testing the outcome we want. This way the test is actually more resilient and if implementation details change but not desired behaviour, then the test doesn't need updating.

Reply:
Thanks @RLavender98 I can see that we're testing the correct template is rendered which is great - can we add a test to make sure that `isTfmPaymentReconciliationFeatureFlagEnabled` is called please?

I appreciate the end result is tested, but something is being called and currently this is not tested. I think this should be tested so that we know the correct `dtfs2-common` feature flag is being called.

Otherwise, if the flag changes in this controller, it could return an incorrect result in the real world; Additionally the test will still pass regardless of which feature flag it's calling.

Reply:
Fair enough, I'll add it then.